Sources/Amplitude/AMPIdentifyInterceptor.m

```diff
@@ -0,0 +1,254 @@
+            NSError *error = nil;
+            NSString *eventJsonString = [AMPEventUtils getJsonString:event eventType:eventType error:&error];
+            // Conversion to JSON string failed, return unmodified event to try to store as a normal identify
+            if (error != nil) {
+                return event;
+            }
+
+            // Store in Intercepted Identify DB
+            [_dbHelper addInterceptedIdentify:eventJsonString];
+
+            // Set timeout for transfer
+            [self scheduleUpload];
+
+            // Event is intercepted, return nil
+            return nil;
+        } else if ([userPropertyOperations objectForKey:AMP_OP_CLEAR_ALL] != nil) {
+           // Clear all pending intercepted Identify's
+           [_dbHelper removeInterceptedIdentifys:[_dbHelper getLastSequenceNumber]];
+        } else {
+           // This is an "active" Identify, merge intercepted user properties
+           [self mergeInterceptedUserProperties:event];
+        }
+    } else if ([eventType isEqualToString:GROUP_IDENTIFY_EVENT]) {
+        // Group identify = no op
+    } else {
+        // Normal event, merge intercepted user properties
+        [self mergeInterceptedUserProperties:event];
+    }
+
+    return event;
+}
+
+// If this returns YES, the given Identify user property operations should be queued and batched later
+- (BOOL)hasInterceptOperationsOnly:(NSDictionary *_Nonnull)userPropertyOperations {
+    NSSet *operations = [NSSet setWithArray:[userPropertyOperations allKeys]];
+    NSSet *interceptSet = [NSSet setWithArray:INTERCEPT_OPS];
+
+    return [operations isSubsetOfSet:interceptSet];
+}
+
+- (void)scheduleUpload {
+    // TODO:
+    if (!_uploadScheduled) {
+        _uploadScheduled = YES;
+        __block __weak AMPIdentifyInterceptor *weakSelf = self;
+        [_backgroundQueue addOperationWithBlock:^{
+            dispatch_async(dispatch_get_main_queue(), ^{
+                [weakSelf performSelector:@selector(uploadCombinedInterceptedIdentify) withObject:nil afterDelay:_interceptedUploadPeriodSeconds];
+            });
+        }];
+    }
+}
+
+- (void)uploadCombinedInterceptedIdentify {
+    NSMutableDictionary *interceptedIdentify = [self getCombinedInterceptedIdentify];
+    if (interceptedIdentify != nil) {
+        NSString *eventType = [AMPEventUtils getEventType:interceptedIdentify];
+
+        NSError *error = nil;
+        NSString *eventJsonString = [AMPEventUtils getJsonString:interceptedIdentify eventType:eventType error:&error];
+        // Conversion to JSON string failed, return unmodified event to try to store as a normal identify
+        if (error == nil) {
+            // Remove pending intercepts from DB
+            [_dbHelper removeInterceptedIdentifys:[_dbHelper getLastSequenceNumber]];
+
+            // Save combined Identify to "active" storage
+            [_dbHelper addIdentify:eventJsonString];
+
+            // Upload immediately
+            [_amplitude uploadEvents];
```

Comment:
Need to make sure we also flush intercepted idenitify on direct calls to uploadEvents()